### PR TITLE
GEOT-5258, Support for storing dates as bigint with postgis.

### PR DESCRIPTION
**Pull request by jdeolive**
Allows for supporting a wider range of dates than the postgresql timestamp type. This is rework of what was originally pull request #997.

### CODE REVIEW

File: modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/BigDate.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2002-2016, Open Source Geospatial Foundation (OSGeo)
```
Comment (jodygarnett) [inline modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/BigDate.java]: New file should be 2016 :)

**Comment by jodygarnett** [comment]
Thanks @jdeolive - this is much cleaner, I like how the type conversion was detected and handled during the visitor.

I would be happy to fix up the headers and merge this in for the beta2 (and lean on the geotools-devel list for testing).

File: modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/BigDate.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.postgis;
+
+import java.util.Date;
+
+/**
+ * Used internally by PostGIS dialect to signal the use of the "bigdate" domain type.
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/BigDate.java]: Wondering, if used internally, does it have to be public, or could it be made package private? I did not actually check if there are impediments, just wondering.

File: modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisFilterToSQL.java [hunk 3]
```diff
@@ -133,7 +134,18 @@ public Object visit(Function function, Object extraData) throws RuntimeException
             throw new RuntimeException(e);
         }
     }
-
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisFilterToSQL.java]: Header not updated

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostgisUDTOnlineTest.java [hunk 4]
```diff
@@ -16,16 +16,30 @@
  */
 package org.geotools.data.postgis;
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostgisUDTOnlineTest.java]: Header not updated

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostgisUDTTestSetup.java [hunk 5]
```diff
@@ -18,6 +18,9 @@
 
 import org.geotools.jdbc.JDBCUDTTestSetup;
 
+import static java.lang.String.format;
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostgisUDTTestSetup.java]: Header not updated

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/ps/PostgisUDTOnlineTest.java [hunk 6]
```diff
@@ -25,7 +25,7 @@
  *
  * @source $URL$
  */
-public class PostgisUDTOnlineTest extends JDBCUDTOnlineTest {
+public class PostgisUDTOnlineTest extends org.geotools.data.postgis.PostgisUDTOnlineTest {
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/ps/PostgisUDTOnlineTest.java]: Header not updated

**Comment by aaime** [comment]
Looks good, found some minor header issues

**Comment by jdeolive** [comment]
Thanks for the review guys. I'll update the headers. We should also be able to make BigDate package protected.

**Pull request merged**